### DISCUSSION TITLE
How can I apply this package to getStaticProps

### DISCUSSION THREAD

**Discussion by golinmarq** [gladly-team/next-firebase-auth#106] [Q&A]
Currently, I'm using Firebase authentication with a session cookie in my NextJS app. The caveat of my app is that I use getInitialProps in my _app.js file so I can use static optimization. 

How can I use this package with SSG? Can I do this? I see that I can use getServerSideProps but it is SSR not SSG

**Comment by kmjennison** [comment]
I'm not sure I understand. If you're using `getInitialProps` in _app.js, then your app will not be static. Every page will be server-rendered.

In the case of getting static props or generating static pages, how are you thinking about using user authentication?

**Reply by golinmarq** [reply]
Yes, that's my issue. I want to change my current logic use getInitialProps and with this it will not be static. 

In the case of getting static props I was wondering if you have an example to apply this package with getStaticProps, I will appreciated

**Reply by kmjennison** [reply]
We don't have plans to support `getInitialProps`. Instead, I recommended using `getServerSideProps` in pages that need it. Related info: #64

Could you explain how you plan to use authentication with `getStaticProps`? We don't currently have an example for this, and I'd like to understand the situation where it's helpful.

**Reply by golinmarq** [reply]
That's my point. Is it possible to do that? I have a ecommerce and the index page will be generated statically according the user but sometimes a user is not logged in.

**Reply by kmjennison** [reply]
Creating a static page for a specific user doesn't make sense to me, but maybe I'm missing something. I expect you'd want either 1) SSR for user-specific pages, or 2) static pages that load the Firebase user only on the client side. This package supports both of those approaches.

If you are hoping to somehow use the authed user when fetching static props, I'd need to better understand your approach to answer your question of whether this package can help. If possible, a demo app or example code would be great.

**Reply by golinmarq** [reply]
You are right, I new on this and I think the approach you gave me is correct. I appreciate your help
